TLM/TLM/U/UIUtil.cs

```diff
@@ -81,5 +83,26 @@ IEnumerable<GameObject> objects
             return s.Replace("[[", UConst.GetKeyboardShortcutColorTagOpener())
                     .Replace("]]", UConst.KEYBOARD_SHORTCUT_CLOSING_TAG);
         }
+
+        /// <summary>
+        /// Replace special markup with UI shortcut color tag (orange).
+        /// Replace wrapped text from 'replacements' of matching index.
+        /// And place closing tag.
+        /// </summary>
+        /// <param name="sourceText">String with keybind wrapped in [[Ctrl]] double square brackets.</param>
+        /// <param name="replacements">Replacements array, each tag match will take own replacement</param>
+        /// <returns>Updated string.</returns>
+        public static string ColorizeDynamicKeybinds([NotNull] string sourceText, string[] replacements) {
+            int i = 0;
+            return new Regex(@"\[\[(.*?)\]\]")
+                .Replace(
+                    sourceText,
+                    new MatchEvaluator(
+                        match => {
+                            return i < replacements.Length
+                                       ? $"{UConst.GetKeyboardShortcutColorTagOpener()}{match.Result(replacements[i++])}{UConst.KEYBOARD_SHORTCUT_CLOSING_TAG}"
```

Comment:
why `match.Result(replacements[i++])` instead of just `replacements[i++]` ?
The documentation of the method says `replacement` not `replacement pattern`.

Reply:
even in your tests you don't use replacement pattern (eg `[$1]`) so I think you `match.Result()` unintentionally.

Reply:
True, returning only `replacement[i++]` should work too. It was late yesterday and I tested a lot of approaches so probably something left 🤔